execution/executor/src/lib.rs

```diff
@@ -40,29 +42,227 @@ use std::{
     collections::{hash_map, HashMap, HashSet},
     convert::TryFrom,
     marker::PhantomData,
-    sync::Arc,
+    sync::{Arc, Mutex, Weak},
 };
 use storage_client::{StorageRead, StorageWrite, VerifiedStateView};
+use storage_proto::StartupInfo;
 use tokio::runtime::Runtime;
 
 static OP_COUNTERS: Lazy<libra_metrics::OpMetrics> =
     Lazy::new(|| libra_metrics::OpMetrics::new_and_registered("executor"));
 
+struct SpeculationBlock {
```

Comment:
I thought the plan was to wrap the SpeculativeBlock in a container outside the core Executor that way we didn't have to modify state sync. I'm not clear why state sync needs speculation at all though... since I thought it would be doing execute and commit?

Reply:
This really should be in a separate module not adding 350 lines to your lib

Reply:
StateSync has nothing to do with `SpeculationBlock`. The plan was to split Executor into two and the one used by state sync will not have this block tree. But I agree the idea of putting it into a standalone module.